### PR TITLE
Add GitHub Actions for OpenBSD, FreeBSD, NetBSD

### PR DESCRIPTION
**Pull request by nao1215**

### CODE REVIEW

File: .github/workflows/platform_test.yml [hunk 1]
```diff
@@ -29,3 +29,61 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: test (freebsd, 1.18)
+        id: test
+        uses: vmactions/freebsd-vm@v0.2.0
+        with:
+          usesh: true
+          prepare: pkg install -y go
+          run: |
+          go mod download
```
[the remark this answers is not in the document]
Comment (nao1215) [reply .github/workflows/platform_test.yml]: Fix [bb49208](https://github.com/nao1215/gup/pull/60/commits/bb4920888295ea5205c735efaffcb461afb9b68b)

File: .github/workflows/platform_test.yml [hunk 2]
```diff
@@ -29,3 +29,63 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: test (illumos, 1.18)
+        id: test
+        uses: papertigers/illumos-vm@r38
+        with:
+          prepare: |
+            pkg install go-118
+          run: |
+          useradd action
```
[the remark this answers is not in the document]
Comment (nao1215) [reply .github/workflows/platform_test.yml]: See [e3f329a](https://github.com/nao1215/gup/pull/60/commits/e3f329a62d847d4b51cc115778dfd8e27fc673ed)

**Comment by nao1215** [comment]
Decide not to support BSD

**Comment by yonas** [comment]
> Decide not to support BSD

@nao1215 Why not?

**Comment by nao1215** [comment]
@yonas 
>Why not? Many people use BSD and gup is a very convenient utility to use.

There are several reasons why I have decided not to support BSD with the gup command:

1. I don't have the time. Since my child was born, my development time has decreased. It is difficult for me to increase the number of supported operating systems on my own.
2. I am not a BSD user. I use Linux and Mac. There is no benefit for me to support BSD. The fact that gup command supports Windows is just a coincidence. I added Windows support to learn cross-platform compatibility.
3. The gup command probably works on BSD. I haven't tested it, though...

**Pull request closed without merging**